File: backend/src/projects/projects.module.ts
import { Module } from '@nestjs/common';
import { PrismaService } from '../prisma/prisma.service';
import { ProjectService } from './services/project.service';
import { TestResultService } from './services/test-result.service';
import { ProjectController } from './controllers/project.controller';
import { TestResultController } from './controllers/test-result.controller';

/**
 * Модуль для управления проектами и результатами тестов
 */
@Module({
  providers: [
    PrismaService,
    ProjectService,
    TestResultService,
  ],
  controllers: [
    ProjectController,
    TestResultController,
  ],
  exports: [
    ProjectService,
    TestResultService,
  ],
})
export class ProjectsModule {}

File: backend/src/projects/ProjectsModule.cs
using Microsoft.Extensions.DependencyInjection;
using Testoria.Projects.Controllers;
using Testoria.Projects.Interfaces;
using Testoria.Projects.Services;

namespace Testoria.Projects
{
    /// <summary>
    /// Модуль для управления проектами и результатами тестов
    /// </summary>
    public sealed class ProjectsModule
    {
        /// <summary>
        /// Регистрирует сервисы модуля проектов
        /// </summary>
        /// <param name="services">Коллекция сервисов</param>
        public static void RegisterServices(IServiceCollection services)
        {
            services.AddScoped<IProjectService, ProjectService>();
            services.AddScoped<TestResultService>();
            services.AddScoped<ProjectController>();
            services.AddScoped<TestResultController>();
        }
    }
}
